WordPress/Classes/Services/PostService.swift

```diff
@@ -0,0 +1,673 @@
+import Foundation
+
+// manages the creation, saving, updating, and merging of Posts, Pages, and associated data such as comments
+class PostService : LocalCoreDataService {
+    static let ErrorDomain = "PostServiceErrorDomain"
+    static let DefaultNumberToSync: UInt = 40
+
+    enum PostType: String {
+        case any = "any"
+        case post = "post"
+        case page = "page"
+
+        func type() -> AbstractPost.Type {
+            switch self {
+            case .post: return Post.self
+            case .page: return Page.self
+            case .any: return AbstractPost.self
+            }
+        }
+    }
+
+    typealias SuccessPosts = ([AbstractPost]) -> Void
+    typealias FailureBasic = (NSError?) -> Void
+    typealias SuccessBasic = () -> Void
+
+    convenience override init() {
+        self.init(managedObjectContext: ContextManager.sharedInstance().mainContext)
+    }
+
+    func makePost(for blog: Blog) -> Post? {
+        return make(type: Post.self, blog: blog)
+    }
+
+    func makePage(for blog: Blog) -> Page? {
+        return make(type: Page.self, blog: blog)
+    }
+
+    func make(postType: PostType?, blog: Blog) -> AbstractPost? {
+        return make(type: postType?.type() ?? Post.self, blog: blog)
+    }
+
+    func make<T: AbstractPost>(type postType: T.Type, blog: Blog) -> T? {
+        assert(self.managedObjectContext == blog.managedObjectContext, "Blog's context should be the the same as the service's")
+
+        guard let post = NSEntityDescription.insertNewObjectForEntityForName(postType.entityName(), inManagedObjectContext: self.managedObjectContext) as? T else {
+            return nil
+        }
+
+        post.blog = blog
+        post.remoteStatus = AbstractPostRemoteStatusSync
+
+        guard let postPost = post as? Post else {
+            return post
+        }
+
+        postPost.postFormat = blog.settings?.defaultPostFormat
+        postPost.postType = Post.typeDefaultIdentifier
+
+        let postCategoryService = PostCategoryService(managedObjectContext:self.managedObjectContext)
+        if let category = postCategoryService.findWithBlogObjectID(blog.objectID, andCategoryID: (blog.settings?.defaultCategoryID)!) {
+            postPost.addCategoriesObject(category)
+        }
+
+        return postPost as? T
+    }
+
+    func makeDraftPost(for blog: Blog) -> Post? {
+        guard let post = self.makePost(for: blog) else { return nil }
+        post.remoteStatus = AbstractPostRemoteStatusLocal
+        return post
+    }
+
+    func makeDraftPage(for blog: Blog) -> Page? {
+        guard let page = self.makePage(for: blog) else { return nil }
+        page.remoteStatus = AbstractPostRemoteStatusLocal
+        return page
+    }
+
+    /**
+     Sync a specific post from the API
+
+     - parameters:
+        - postID: The ID of the post to sync
+        - blog: The blog that has the post
+        - success: A success block
+        - failure: A failure block
+     */
```

Comment:
Please, check out the *Swift* styled comments we're using (IE Notification.swift should be full of it). Should look like this:

```
/// Some method description
/// - parameters:
///  - somethingID: some description
///
```